contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -3,6 +3,20 @@
 #include "funcapi.h"
 #include "pgstat.h"
 
+#include "postgres.h"
+#include "access/hash.h"
+#include "utils/builtins.h"
+#include "utils/date.h"
+#include "utils/datetime.h"
+#include "libpq/pqformat.h"
+#include "utils/timestamp.h"
+
+#include "fmgr.h"
+#include "miscadmin.h"
+#include "../../babelfishpg_common/src/datetimeoffset.h"
```

Comment:
ideally, avoid including cross extension headers. You rendezvous hooks instead

Reply:
This PR is still in progress, want Parikshit to get datepart working first.
Also including header file is not a problem since it creates a copy of the header and places it with TSQL build extension

Reply:
Including header files defeats the purpose of extensions. Extension functionality can be invoked by hooks and replaced with different implementation when desired. Including headers creates hard dependency.